modules/core/04-channel/v2/keeper/events.go

```diff
@@ -14,99 +16,131 @@ import (
 func emitSendPacketEvents(ctx context.Context, packet types.Packet) {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 
+	encodedPacket, err := proto.Marshal(&packet)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
When event emission is changed to env event service this func will be a method on `Keeper`. Will have access to codec on method recvr then instead of plumbing it through now and can just use `k.cdc.MustMarshal(packet)`

I can create an issue. I don't want to block on this going through

Reply:
https://github.com/cosmos/ibc-go/issues/7614